src/main/java/dev/task/dndquest/security/authentication/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package dev.task.dndquest.security.authentication;
+
+import dev.task.dndquest.exception.PlayerNotFoundException;
+import dev.task.dndquest.model.dto.AuthenticationResponseDto;
+import dev.task.dndquest.model.dto.PlayerRequestDto;
+import dev.task.dndquest.model.entity.Player;
+import dev.task.dndquest.security.jwt.JwtProvider;
+import dev.task.dndquest.service.PlayerService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private final PlayerService playerService;
+    private final PasswordEncoder passwordEncoder;
+    private final JwtProvider jwtProvider;
+
+    @Override
+    public AuthenticationResponseDto login(PlayerRequestDto dto)
+            throws PlayerNotFoundException{
+        Player player = playerService.findByLogin(dto.getLogin());
+        if (!passwordEncoder.matches(dto.getPassword(), player.getPassword())) {
+            throw new PlayerNotFoundException();

```

Comment:
??

Reply:
.findByLogin() can throw exception, thats why it marked as throws. In line 25 it checks if dto has the same pass with player from db, and if no - throw exception. 
Or question in something else?

Reply:
qstn is - why PlayerNotFound if password missmatched? It is generally wrong - all unsuccessful attempt to login must result in BadCredentialsException

Reply:
i got u.